Assets/Scripts/Projectiles/EnemyProjectile.cs

```diff
@@ -0,0 +1,52 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class EnemyProjectile : MonoBehaviour
+{
+    private Vector3 _firingPoint;
+    [SerializeField] private float _projectileSpeed;
+
+    [SerializeField] private float _maxProjectileDistance;
+
+    [SerializeField] private float _damage;
+
+    private GameObject _triggeringPlayer;
+
+    private void Start()
+    {
+        _firingPoint = transform.position;
+    }
+
+    private void Update()
+    {
+        
+        MoveProjectile();
+    }
+
+    void MoveProjectile() {
+        if (Vector3.Distance(_firingPoint, transform.position) > _maxProjectileDistance)
+        {
+            Destroy(this.gameObject);
+        } 
+        else 
+        {
+            transform.Translate(Vector3.forward * _projectileSpeed * Time.deltaTime);
+        }
+        transform.Translate(Vector3.forward * _projectileSpeed * Time.deltaTime);
```

Comment:
Ai EXACT aceeași linie și în `else` și după `else`. De ce?

Ordinea înmulțirilor e ineficientă. Mai bine întâi se înmulțesc toți scalarii și apoi se înmulțește rezultatul cu vectorul ăla.